parquet-column/src/main/java/org/apache/parquet/column/statistics/Statistics.java

```diff
@@ -175,10 +200,47 @@ public void mergeStatistics(Statistics stats) {
    */
   abstract public void setMinMaxFromBytes(byte[] minBytes, byte[] maxBytes);
 
+  /**
+   * Returns the generic object representing the min value in the statistics. The self comparing logic of the returned
+   * object might not be the proper one (e.g. unsigned comparison for int/long) therefore it is strongly recommended to
+   * use the related comparing method {@link #compareToMin(Object)} or the comparator returned by {@link
+   * #comparator()}.
+   */
   abstract public T genericGetMin();
+
+  /**
+   * Returns the generic object representing the max value in the statistics. The self comparing logic of the returned
```

Comment:
A few clarifications:
* Instead of "generic object" just say "max value". The type is clear from the use of generics. If anything, add a `@param <T>` to the class-level javadoc.
* Instead of "self comparing logic", use "Java natural order" or reference `compareTo` to be explicit.
* Instead of "strongly recommended", I think you should explain why: the sort order for logical types may not be the order produced by `compareTo`. Users should use the logical type's defined order and can use `#comparator()` to get a `Comparator` instance that implements that order.